tests/integration/test_bundle.py

```diff
@@ -68,45 +110,64 @@ async def test_render_and_deploy_bundle(ops_test: OpsTest):
 
 
 @pytest.mark.abort_on_fail
-async def test_hydra_is_up(ops_test: OpsTest):
+async def test_hydra_is_up(ops_test: OpsTest) -> None:
     """Check that hydra and its environment dependencies (e.g. the database) are responsive."""
-    app_name = "hydra"
-    admin_address = await get_unit_address(ops_test, TRAEFIK_ADMIN_APP, 0)
+    hydra_url = await get_reverse_proxy_app_url(ops_test, APPS.TRAEFIK_ADMIN, APPS.HYDRA)
 
-    health_check_url = f"http://{admin_address}/{ops_test.model.name}-{app_name}/health/ready"
-    logger.info(f"Hydra admin health check address: {health_check_url}")
+    health_check_url = join(hydra_url, "health/ready")
 
-    resp = requests.get(health_check_url)
+    resp = requests.get(health_check_url, verify=False)
     assert resp.status_code == 200
 
 
 @pytest.mark.abort_on_fail
-async def test_kratos_is_up(ops_test: OpsTest):
+async def test_kratos_is_up(ops_test: OpsTest) -> None:
     """Check that kratos and its environment dependencies (e.g. the database) are responsive."""
-    app_name = "kratos"
-    admin_address = await get_unit_address(ops_test, TRAEFIK_ADMIN_APP, 0)
+    kratos_url = await get_reverse_proxy_app_url(ops_test, APPS.TRAEFIK_ADMIN, APPS.KRATOS)
 
-    health_check_url = (
-        f"http://{admin_address}/{ops_test.model.name}-{app_name}/admin/health/ready"
-    )
-    logger.info(f"Kratos admin health check address: {health_check_url}")
+    health_check_url = join(kratos_url, "admin/health/ready")
 
-    resp = requests.get(health_check_url)
+    resp = requests.get(health_check_url, verify=False)
     assert resp.status_code == 200
 
 
-async def test_multiple_kratos_external_idp_integrators(ops_test: OpsTest):
+@pytest.mark.abort_on_fail
+async def test_kratos_external_idp_redirect_url(ops_test: OpsTest, client: Client) -> None:
+    get_redirect_uri_action = (
+        await ops_test.model.applications[APPS.KRATOS_EXTERNAL_IDP_INTEGRATOR]
+        .units[0]
+        .run_action("get-redirect-uri")
+    )
+
+    action_output = await get_redirect_uri_action.wait()
+    assert "redirect-uri" in action_output.results
+
+    apply_dex_resources(
```

Comment:
Nitpick:

- I see there are some `dex`-specific things existing in the test script, like `dex_user_email`, `dex_user_password`, `apply_dex_resources`, etc. Do you think we could rename them or isolate them in the `conftest.py`? So that the test cases only have the knowledge of the external IDP service, not a specific implementation. I guess in our case, the fixtures in the `conftest.py` are used as the isolation layer.

Reply:
Ok, I renamed them.